general/project/repoutils.go

```
@@ -0,0 +1,137 @@
+	}
+	params := services.NewLocalRepositoryBaseParams()
+	params.PackageType = string(technologyType)
+	params.Key = RepoDefaultName[technologyType][Local]
+	if repoExists(servicesManager, params.Key) {
+		return nil
+	}
+	return servicesManager.CreateLocalRepositoryWithParams(params)
+}
+
+func CreateDefaultRemoteRepo(technologyType coreutils.Technology, serverId string) error {
+	servicesManager, err := getServiceManager(serverId)
+	if err != nil {
+		return err
+	}
+	params := services.NewRemoteRepositoryBaseParams()
+	params.PackageType = string(technologyType)
+	params.Key = RepoDefaultName[technologyType][Remote]
+	params.Url = RepoDefaultName[technologyType][RemoteUrl]
+	if repoExists(servicesManager, params.Key) {
+		return nil
+	}
+	return servicesManager.CreateRemoteRepositoryWithParams(params)
+}
+
+func CreateDefaultVirtualRepo(technologyType coreutils.Technology, serverId string) error {
+	servicesManager, err := getServiceManager(serverId)
+	if err != nil {
+		return err
+	}
+	params := services.NewVirtualRepositoryBaseParams()
+	params.PackageType = string(technologyType)
+	params.Key = RepoDefaultName[technologyType][Virtual]
+	params.Repositories = []string{RepoDefaultName[technologyType][Local], RepoDefaultName[technologyType][Remote]}
+	params.DefaultDeploymentRepo = RepoDefaultName[technologyType][Local]
+	if repoExists(servicesManager, params.Key) {
+		return nil
+	}
+	return servicesManager.CreateVirtualRepositoryWithParams(params)
+}
+
+func getServiceManager(serverId string) (artifactory.ArtifactoryServicesManager, error) {
+	serviceDetails, err := config.GetSpecificConfig(serverId, true, false)
+	if err != nil {
+		return nil, err
+	}
+	return rtUtils.CreateServiceManager(serviceDetails, -1, false)
+
+}
+
+// Check if default repository is allready exists
+func repoExists(servicesManager artifactory.ArtifactoryServicesManager, repoKey string) bool {
+	repo := &services.RepositoryDetails{}
+	servicesManager.GetRepository(repoKey, repo)
```

Let's not ignore the error returned by this method.